roles/postgres/tasks/restore.yml

```diff
@@ -0,0 +1,33 @@
+- name: Restore database backups if necessary
+  become: true
+  become_user: postgres
+  block:
+    - name: Check if we need to restore a database backup
+      community.postgresql.postgresql_query:
+        db: "{{ item.database }}"
+        query: "SELECT * FROM pg_catalog.pg_tables WHERE tableowner = %s"
```

Comment:
What if the user we have owns multiple databases? Is there any way to check if there are any tables in the database instead of checking owned tables number across the server?

Reply:
Oh, it's grand: `pg_catalog.pg_tables` is a magical system view that exists in all databases, and all the information is local to a particular database.

Reply:
... and we connect to the database above.